File: s3GoSDK/sample/GOSDK/src/DeleteObject.go
package main

import (
	"context"
	"fmt"

	"github.com/aws/aws-sdk-go-v2/service/s3"
)

func DeleteObject(arg [3]string, filePath [3]string) {
	input := &s3.DeleteObjectInput{
		Bucket: &arg[0],
		Key:    &filePath[0],
	}

	_, deleteobject_err := Client.DeleteObject(context.TODO(), input)
	if deleteobject_err != nil {
		fmt.Println("Got an error deleting item:")
		fmt.Println(deleteobject_err)
		return
	}
	fmt.Println("Deleted " + filePath[0] + " from " + arg[0])
}

File: s3RustSDK/sample/src/delete_object.rs
use s3::Client;
use std::process;

pub async fn deleteobject(client: Client, bucket: String, key: String) {
    match client
        .delete_object()
        .bucket(&bucket)
        .key(&key)
        .send()
        .await
    {
        Ok(_) => {
            println!("Deleted object {}/{}\n", bucket, key);
        }
        Err(e) => {
            println!("Got an error deleting object");
            println!("{}", e);
            process::exit(1);
        }
    };
}
